test/NuGet.Core.Tests/NuGet.Protocol.Tests/Providers/ReadmeUriTemplateResourceProviderTests.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using Newtonsoft.Json.Linq;
+using NuGet.Configuration;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+using Xunit;
+
+namespace NuGet.Protocol.Tests.Providers
+{
+    public class ReadmeUriTemplateResourceProviderTests
+    {
+        private const string ResourceType = "ReadmeUriTemplate/6.13.0";
+
+        private readonly PackageSource _packageSource;
+        private readonly ReadmeUriTemplateResourceProvider _target;
+
+        public ReadmeUriTemplateResourceProviderTests()
+        {
+            _packageSource = new PackageSource("https://unit.test");
+            _target = new ReadmeUriTemplateResourceProvider();
+        }
+
+        [Fact]
+        public async Task TryCreate_WhenResourceDoesNotExist_ReturnsNull()
+        {
+            var resourceProviders = new ResourceProvider[]
+            {
+                CreateServiceIndexResourceV3Provider(),
+                _target
+            };
+            var sourceRepository = new SourceRepository(_packageSource, resourceProviders);
+
+            Tuple<bool, INuGetResource> result = await _target.TryCreate(sourceRepository, CancellationToken.None);
+
+            Assert.False(result.Item1);
+            Assert.Null(result.Item2);
+        }
+
+        [Fact]
+        public async Task TryCreate_WhenResourceExists_ReturnsValidResourceAsync()
+        {
+            var serviceEntry = new RawServiceIndexEntry("https://unit.test/packages/{lower_id}/{lower_version}/readme", ResourceType);
+            var resourceProviders = new ResourceProvider[]
+            {
+                CreateServiceIndexResourceV3Provider(serviceEntry),
+                _target
+            };
+            var sourceRepository = new SourceRepository(_packageSource, resourceProviders);
+
+            Tuple<bool, INuGetResource> result = await _target.TryCreate(sourceRepository, CancellationToken.None);
+
+            Assert.True(result.Item1);
+            Assert.IsType<ReadmeUriTemplateResource>(result.Item2);
+            Assert.Equal(
+                "https://unit.test/packages/mypackage/1.0.0/readme",
+                ((ReadmeUriTemplateResource)result.Item2).GetReadmeUrl("MyPackage", NuGetVersion.Parse("1.0.0")));
```

Comment:
nit: This feels like an Act to me, and also would be good to include `GetReadmeUrl` in the test title somewhere. eg, `TryCreate_WhenResourceExists_GetReadmeUrlIsCorrect` or `GetReadmeUrl_WhenReadmeUriTemplateResourceExists_ReturnsCorrrectUrl`

Reply:
Hmm I'm going to delete that line all together. ReadmeUriTemplateResourceTests already tests that method.